enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -10075,69 +10075,77 @@ class AdjointGenerator
+                                            orig->getCalledOperand(), args,
+                                            orig->getName() + "'mi");
 #else
-              anti = bb.CreateCall(orig->getCalledValue(), args,
-                                   orig->getName() + "'mi");
+                anti = bb.CreateCall(orig->getCalledValue(), args,
+                                     orig->getName() + "'mi");
 #endif
-              cast<CallInst>(anti)->setAttributes(orig->getAttributes());
-              cast<CallInst>(anti)->setCallingConv(orig->getCallingConv());
-              cast<CallInst>(anti)->setTailCallKind(orig->getTailCallKind());
-              cast<CallInst>(anti)->setDebugLoc(dbgLoc);
+                cast<CallInst>(anti)->setAttributes(orig->getAttributes());
+                cast<CallInst>(anti)->setCallingConv(orig->getCallingConv());
+                cast<CallInst>(anti)->setTailCallKind(orig->getTailCallKind());
+                cast<CallInst>(anti)->setDebugLoc(dbgLoc);
 
 #if LLVM_VERSION_MAJOR >= 14
-              cast<CallInst>(anti)->addAttributeAtIndex(
-                  AttributeList::ReturnIndex, Attribute::NoAlias);
-              cast<CallInst>(anti)->addAttributeAtIndex(
-                  AttributeList::ReturnIndex, Attribute::NonNull);
+                cast<CallInst>(anti)->addAttributeAtIndex(
+                    AttributeList::ReturnIndex, Attribute::NoAlias);
+                cast<CallInst>(anti)->addAttributeAtIndex(
+                    AttributeList::ReturnIndex, Attribute::NonNull);
 #else
-              cast<CallInst>(anti)->addAttribute(AttributeList::ReturnIndex,
-                                                 Attribute::NoAlias);
-              cast<CallInst>(anti)->addAttribute(AttributeList::ReturnIndex,
-                                                 Attribute::NonNull);
+                cast<CallInst>(anti)->addAttribute(AttributeList::ReturnIndex,
+                                                   Attribute::NoAlias);
+                cast<CallInst>(anti)->addAttribute(AttributeList::ReturnIndex,
+                                                   Attribute::NonNull);
 #endif
+                return anti;
+              };
+
+              anti = applyChainRule(orig->getType(), bb, rule);
 
               if (called->getName() == "malloc" ||
                   called->getName() == "_Znwm") {
                 if (auto ci = dyn_cast<ConstantInt>(args[0])) {
                   unsigned derefBytes = ci->getLimitedValue();
                   CallInst *cal =
                       cast<CallInst>(gutils->getNewFromOriginal(orig));
+                  auto rule = [&](Value *anti) {
 #if LLVM_VERSION_MAJOR >= 14
-                  cast<CallInst>(anti)->addDereferenceableRetAttr(derefBytes);
-                  cal->addDereferenceableRetAttr(derefBytes);
+                    cast<CallInst>(anti)->addDereferenceableRetAttr(derefBytes);
```

Comment:
This is not something you want to do an applychainrules for, as you actually need this to apply to the actual call itself. In retrospect the extractMeta I added, might actually give you the correct behavior on this, but I would still prefer this functionality be inside of the first applyChainRule call all together.